Classes/FrontendEnvironment/Tsfe.php

```diff
@@ -151,6 +152,8 @@ protected function initializeTsfe(int $pageId, int $language = 0, ?int $rootPage
             if (!empty($pageRecord['fe_group'])) {
                 $userGroups = array_unique(array_merge($userGroups, explode(',', $pageRecord['fe_group'])));
```

Comment:
```suggestion
                $userGroups = explode(',', $pageRecord['fe_group']);
```
IMHO the -1 group isn't relevant if other groups are set.
the [0] group has each user.